pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -347,55 +375,57 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             IERC4626 wrappedToken = IERC4626(address(erc4626PoolTokens[i]));
             IERC20 underlyingToken = IERC20(_vault.getBufferAsset(wrappedToken));
 
-            // If the Vault returns address 0 as underlying, it means that the ERC4626 token buffer was not
-            // initialized. Thus, the Router treats it as a non-ERC4626 token.
-            if (address(underlyingToken) == address(0)) {
+            if (useWrappedTokens[i] == true) {
+                amountsIn[i] = amountsInRaw[i];
+
                 if (amountsIn[i] > params.maxAmountsIn[i]) {
```

Comment:
One thing that I'm realizing is that we have `maxAmountsIn` and also `limits` here. This is because in add unbalanced we don't really need to apply limits.

But truth is that it doesn't really matter. For add unbalanced we fix the inputs to max, so this check won't fail. In other words, we can get rid of the limits argument. I'll do a draft; that's out of scope of this PR.

Reply:
hmm that's not the case actually. I'll double check.

Reply:
ok so my confusion here is that limits is actually `minAmountsOut` whenever we need to wrap exact in, and matches `maxAmountsIn` when it's exact out and we need to wrap.

I'm not liking the multiple meanings of things. It's just too tricky to follow. Let's think it through a bit further (this would be out of scope of this PR anyways).

Reply:
Perhaps applying the limit unconditionally at the end is a good start.
But we still need to handle the limits in the wrap operation, which has different meanings when given in / given out.